File: src/main/java/io/vproxy/vfx/ui/pane/TransparentContentFusionPane.java
package io.vproxy.vfx.ui.pane;

public class TransparentContentFusionPane extends TransparentFusionPane {
    public TransparentContentFusionPane() {
        super();
    }

    public TransparentContentFusionPane(boolean manuallyHandleOuterRegion) {
        super(manuallyHandleOuterRegion);
    }

    @Override
    protected AbstractFusionPane buildRootNode() {
        return new TransparentContentFusionPaneImpl();
    }

    protected class TransparentContentFusionPaneImpl extends TransparentFusionPaneImpl {
        @Override
        protected double normalContentOpacity() {
            return 0.5;
        }
    }
}

File: src-ts/src/ui/pane/TransparentContentFusionPane.ts
import { TransparentFusionPane } from './TransparentFusionPane.js';
import { TransparentFusionPaneImpl } from './TransparentFusionPane.js';

export class TransparentContentFusionPane extends TransparentFusionPane {
  constructor();
  constructor(manuallyHandleOuterRegion: boolean);
  constructor(manuallyHandleOuterRegion?: boolean) {
    super(manuallyHandleOuterRegion ?? true);
  }

  protected override buildRootNode() {
    return new TransparentContentFusionPaneImpl(this);
  }
}

export class TransparentContentFusionPaneImpl extends TransparentFusionPaneImpl {
  override normalContentOpacity(): number {
    return 0.5;
  }
}
